enarx-keep-sev-shim/src/gdt.rs

```diff
@@ -18,8 +18,89 @@ use x86_64::{PrivilegeLevel, VirtAddr};
 const STACK_NUM_PAGES: usize = 5;
 static mut STACKS: [[Page; STACK_NUM_PAGES]; 7] = [[Page::zeroed(); STACK_NUM_PAGES]; 7];
 
-pub(crate) static mut TSS: TaskStateSegment = TaskStateSegment::new();
-pub(crate) static mut GDT: GlobalDescriptorTable = GlobalDescriptorTable::new();
+/// FIXME: will be replaced by a dynamically allocated stack with safe guards
+pub static mut LEVEL_0_STACK: [Page; 5] = [Page::zeroed(); 5];
+
+lazy_static! {
+    /// The global TSS
+    pub static ref TSS: TaskStateSegment = {
+        let mut tss = TaskStateSegment::new();
+
+        #[inline(always)]
+        #[allow(clippy::integer_arithmetic)]
+        fn stack_end(stack_start: VirtAddr, num_pages: usize) -> VirtAddr {
+            stack_start + num_pages * Page::size()
+        }
+
+        unsafe {
+            tss.privilege_stack_table[0] = VirtAddr::from_ptr(&LEVEL_0_STACK);
+        }
+
+        // Assign the stacks for the exceptions and interrupts
+        unsafe {
+            tss.interrupt_stack_table
+                .iter_mut()
+                .zip(&STACKS)
+                .for_each(|(p, v)| *p = stack_end(VirtAddr::from_ptr(v), STACK_NUM_PAGES));
+
+            debug_assert_eq!(STACKS.len(), tss.interrupt_stack_table.len());
+        }
+
+        tss
+    };
+}
+
+/// The Selectors used in the GDT setup
+pub struct Selectors {
```

Comment:
I think the `_sel` suffix on these fields should be removed, since I think their intent is already well-described by accessing a field of a struct called `Selectors`.

Reply:
done